dotCMS/src/main/java/com/dotmarketing/util/ImportStarterUtil.java

```diff
@@ -0,0 +1,1148 @@
+     * @param out - Printwriter to write responses to Reponse Printwriter so this method can write to
+     *        screen.
+     */
+    private void doXMLFileImport(File f) throws Exception {
+        doXMLFileImport(f, null);
+    }
+    
+    
+    
+
+    /**
+     * This method takes an XML file and will try to import it via XStream and Hibernate.
+     *
+     * @param f - File to be parsed and imported.
+     * @param out - Printwriter to write responses to Reponse Printwriter so this method can write to
+     *        screen.
+     * @param filter
+     * @throws DotDataException
+     * @throws HibernateException
+     * @throws EncryptorException
+     * @throws SQLException 
+     * @throws DotSecurityException 
+     * @throws DuplicateUserException 
+     * @throws NoSuchMethodException 
+     * @throws InvocationTargetException 
+     * @throws IllegalAccessException 
+     */
+    private void doXMLFileImport(File f, ObjectFilter filter) throws Exception {
+        if (f == null) {
+            return;
+        }
+
+
+
+            Class _importClass = null;
+            HibernateUtil _dh = null;
+
+            boolean counter = false;
+            boolean image = false;
+            boolean portlet = false;
+            boolean logsMappers = false;
+
+
+            Pattern classNamePattern = Pattern.compile("_[0-9]{8}");
+            Logger.info(this, "**** Importing the file: " + f + " *****");
+
+            /* if we have a multipart import file */
+            
+            final String _className = classNamePattern.matcher(f.getName()).find() 
+                   ?  f.getName().substring(0, f.getName().lastIndexOf("_"))
+                   :   f.getName().substring(0, f.getName().lastIndexOf("."));
+
+
+            if (_className.equals("Counter")) {
+                counter = true;
+            } else if (_className.equals("Image")) {
+                image = true;
+            } else if (_className.equals("Portlet")) {
+                portlet = true;
+            } else if (_className.equals("LogsMappers")) {
+                logsMappers = true;
+            } else {
+                try {
+                    _importClass = Class.forName(_className);
+                } catch (Exception e) {
+                    Logger.error(this, "Class not found " + _className);
+                    return;
+                }
+            }
+
+            Logger.info(this, "Importing:\t" + _className);
+
+
+            List l = new ArrayList();
+
+            List all = (List) BundlerUtil.xmlToObject(f);
+            if (filter != null) {
+                for (Object obj : all)
+                    if (filter.includeIt(obj))
+                        l.add(obj);
+            } else {
+                l = all;
+            }
+
+
+            Logger.info(this, "Found :\t" + l.size() + " " + _className + "(s)");
+            if (counter) {
+                for (int j = 0; j < l.size(); j++) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This for loop can be replaced by a foreach loop](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)